goal_src/engine/ambient/mood-tables.gc

```diff
@@ -5,3 +5,4106 @@
+  )
+
+;; definition for function update-mood-shadow-direction
+;; INFO: Return type mismatch int vs none.
+(defun update-mood-shadow-direction ((arg0 mood-lights))
+  (let ((v1-0 (-> arg0 shadow)))
+   (set! (-> v1-0 x) (- (-> arg0 direction x)))
+   (set! (-> v1-0 y) (- (-> arg0 direction y)))
+   (set! (-> v1-0 z) (- (-> arg0 direction z)))
+   (if (< (-> arg0 direction y) (l.f L116))
+    (let* ((f0-7 (l.f L103))
+           (f1-1 (-> v1-0 x))
+           (f1-3 (* f1-1 f1-1))
+           (f2-0 (-> v1-0 z))
+           (f0-8 (/ f0-7 (sqrtf (+ f1-3 (* f2-0 f2-0)))))
+           )
+     (set! (-> v1-0 x) (* (-> v1-0 x) f0-8))
+     (set! (-> v1-0 y) (l.f L118))
+     (set! (-> v1-0 z) (* (-> v1-0 z) f0-8))
+     )
+    )
+   )
+  (let ((v0-0 0))
+   )
+  (none)
+  )
+
+;; definition for function update-mood-erase-color
+;; INFO: Return type mismatch int vs none.
+(defun update-mood-erase-color ((arg0 mood-fog) (arg1 mood-lights))
+  (let ((s5-0 (-> arg0 erase-color)))
+   (vector+float*!
+    s5-0
+    (-> arg1 amb-color)
+    (-> arg1 lgt-color)
+    (-> arg1 direction y)
+    )
+   (vector*! s5-0 s5-0 (-> *ocean-map-village2* far-color))
+   (set! (-> s5-0 x) (* (l.f L105) (-> s5-0 x)))
+   (set! (-> s5-0 y) (* (l.f L105) (-> s5-0 y)))
+   (set! (-> s5-0 z) (* (l.f L105) (-> s5-0 z)))
+   (let ((f0-8 (* (l.f L89) (- (l.f L102) (-> arg0 fog-dists w)))))
+    (vector4-lerp! s5-0 s5-0 (-> arg0 fog-color) f0-8)
+    )
+   )
+  (let ((v0-3 0))
+   )
+  (none)
+  )
+
+;; definition for function update-mood-erase-color2
+;; INFO: Return type mismatch int vs none.
+;; Used lq/sq
+(defun
+  update-mood-erase-color2
+  ((arg0 mood-fog) (arg1 mood-lights) (arg2 mood-lights))
+  (let ((s5-0 (-> arg0 erase-color)))
+   (let ((s4-0 (new 'stack-no-clear 'mood-fog)))
+    (set! (-> s4-0 fog-color quad) (the-as uint128 0))
+    (vector+float*!
+     s5-0
+     (-> arg1 amb-color)
+     (-> arg1 lgt-color)
+     (-> arg1 direction y)
+     )
+    (vector*! s5-0 s5-0 (-> *ocean-map-village2* far-color))
+    (set! (-> s5-0 x) (* (l.f L105) (-> s5-0 x)))
+    (set! (-> s5-0 y) (* (l.f L105) (-> s5-0 y)))
+    (set! (-> s5-0 z) (* (l.f L105) (-> s5-0 z)))
+    (vector+float*!
+     (the-as vector s4-0)
+     (-> arg2 amb-color)
+     (-> arg2 lgt-color)
+     (-> arg2 direction y)
+     )
+    (vector*!
+     (the-as vector s4-0)
+     (the-as vector s4-0)
+     (-> *ocean-map-village2* far-color)
+     )
+    (set! (-> s4-0 fog-color x) (* (l.f L105) (-> s4-0 fog-color x)))
+    (set! (-> s4-0 fog-color y) (* (l.f L105) (-> s4-0 fog-color y)))
+    (set! (-> s4-0 fog-color z) (* (l.f L105) (-> s4-0 fog-color z)))
+    (vector4-lerp! s5-0 s5-0 (the-as vector s4-0) 0.5)
+    )
+   (let ((f0-15 (* (l.f L89) (- (l.f L102) (-> arg0 fog-dists w)))))
+    (vector4-lerp! s5-0 s5-0 (-> arg0 fog-color) f0-15)
+    )
+   )
+  (let ((v0-6 0))
+   )
+  (none)
+  )
+
+;; definition for symbol *default-interp-table*, type sky-color-day
+(define
+  *default-interp-table*
```

Comment:
I'd recommend putting `define` and `*default-interp-table*` on the same line here and in all the other defines.